x-pack/plugins/kubernetes_security/public/components/count_widget/index.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiText, EuiLoadingSpinner, EuiIconTip } from '@elastic/eui';
+import { useStyles } from './styles';
+import type { IndexPattern, GlobalFilter } from '../../types';
+import { addTimerangeToQuery } from '../../utils/add_timerange_to_query';
+import { useFetchCountWidgetData } from './hooks';
+import { addResourceTypeToFilterQuery, numberFormatter } from './helpers';
+
+export const LOADING_TEST_ID = 'kubernetesSecurity:count-widget-loading';
+export const TOOLTIP_TEST_ID = 'kubernetesSecurity:count-widget-tooltip';
+
+export interface CountWidgetDeps {
+  title: string;
+  widgetKey: string;
+  indexPattern?: IndexPattern;
+  globalFilter: GlobalFilter;
+  groupedBy: string;
+}
+
+export const CountWidget = ({
+  title,
+  widgetKey,
+  indexPattern,
+  globalFilter,
+  groupedBy,
+}: CountWidgetDeps) => {
+  const styles = useStyles();
+
+  const filterQueryWithTimeRange = useMemo(() => {
+    let globalFilterModified = globalFilter.filterQuery;
+
+    if (widgetKey === 'CountNodesWidgets') {
+      globalFilterModified = addResourceTypeToFilterQuery(globalFilter.filterQuery, 'node');
+    }
+
+    if (widgetKey === 'CountPodsWidgets') {
+      globalFilterModified = addResourceTypeToFilterQuery(globalFilter.filterQuery, 'pod');
+    }
+    return addTimerangeToQuery(globalFilterModified, globalFilter.startDate, globalFilter.endDate);
+  }, [globalFilter.filterQuery, globalFilter.startDate, globalFilter.endDate, widgetKey]);
+
+  const { data, isLoading } = useFetchCountWidgetData(
+    widgetKey,
+    filterQueryWithTimeRange,
+    groupedBy,
+    indexPattern?.title
+  );
+
+  const countValue = useMemo((): number => {
+    return data ? Number(data?.pages[0]) : 0;
```

Comment:
I believe you could just do `data?.pages[0].value` instead of just casting it to a number, right? If you cast `data?.pages[0]` (i assume its a `CountResult` object) to a number, it's going to return `NaN`